pkg/solana/logpoller/log_poller.go

```diff
@@ -44,15 +66,93 @@ func New(lggr logger.SugaredLogger, orm ORM) *LogPoller {
 		Start: lp.start,
 	}.NewServiceEngine(lggr)
 	lp.lggr = lp.eng.SugaredLogger
+
 	return lp
 }
 
 func (lp *LogPoller) start(context.Context) error {
+	cl, err := lp.client.Get()
+	if err != nil {
+		return err
+	}
+	lp.collector = NewEncodedLogCollector(cl, lp, lp.lggr)
+
 	lp.eng.Go(lp.run)
 	lp.eng.Go(lp.backgroundWorkerRun)
 	return nil
 }
 
+func makeLogIndex(txIndex int, txLogIndex uint) int64 {
+	if txIndex < 0 || txIndex > math.MaxUint32 || txLogIndex > math.MaxUint32 {
+		panic(fmt.Sprintf("txIndex or txLogIndex out of range: txIndex=%d, txLogIndex=%d", txIndex, txLogIndex))
+	}
+	return int64(math.MaxUint32*uint32(txIndex) + uint32(txLogIndex))
+}
+
+// Process - process stream of events coming from log ingester
+func (lp *LogPoller) Process(programEvent ProgramEvent) (err error) {
+	ctx, cancel := utils.ContextFromChan(lp.eng.StopChan)
+	defer cancel()
+
+	blockData := programEvent.BlockData
+
+	matchingFilters := lp.filters.MatchingFiltersForEncodedEvent(programEvent)
+	if matchingFilters == nil {
+		return
+	}
+
+	var logs []Log
+	for filter := range matchingFilters {
+		log := Log{
+			FilterID:       filter.ID,
+			ChainID:        lp.orm.ChainID(),
+			LogIndex:       makeLogIndex(blockData.TransactionIndex, blockData.TransactionLogIndex),
+			BlockHash:      Hash(blockData.BlockHash),
+			BlockNumber:    int64(blockData.BlockHeight),
+			BlockTimestamp: blockData.BlockTime.Time().UTC(),
+			Address:        filter.Address,
+			EventSig:       filter.EventSig,
+			TxHash:         Signature(blockData.TransactionHash),
+		}
+
+		log.Data, err = base64.StdEncoding.DecodeString(programEvent.Data)
```

Comment:
errors related to decoding should be logged as critical as LP won't be able to ever process the msg. 
We might also want just to abandon this log instead of constantly retrying, as we may be able to process other msgs.